src/main/scala/com/horizon/exchangeapi/ServicesRoutes.scala

```diff
@@ -1096,6 +1103,46 @@ trait ServiceRoutes extends ScalatraBase with FutureSupport with SwaggerSupport
     })
   })
 
+  // =========== POST /orgs/{orgid}/services/{service}/search  ===============================
+
+  val postServiceSearch =
+    (apiOperation[PostServiceSearchResponse]("postServiceSearch")
+      summary("Returns the nodes a service is running on")
+      description """Returns a list of all the nodes a service is running on."""
+      parameters(
+      Parameter("orgid", DataType.String, Option[String]("Organization id."), paramType=ParamType.Path),
+      Parameter("id", DataType.String, Option[String]("Username of exchange user, or ID of an agbot. This parameter can also be passed in the HTTP Header."), paramType=ParamType.Query, required=false),
+      Parameter("token", DataType.String, Option[String]("Password of exchange user, or token of the agbot. This parameter can also be passed in the HTTP Header."), paramType=ParamType.Query, required=false),
+    )
+      responseMessages(ResponseMessage(HttpCode.POST_OK,"created/updated"), ResponseMessage(HttpCode.BADCREDS,"invalid credentials"), ResponseMessage(HttpCode.ACCESS_DENIED,"access denied"), ResponseMessage(HttpCode.BAD_INPUT,"bad input"), ResponseMessage(HttpCode.NOT_FOUND,"not found"))
+      )
+  val postServiceSearch2 = (apiOperation[PostNodeHealthRequest]("postSearchNodeHealth2") summary("a") description("a"))
+
+  /** Called by the agbot to get recent info about nodes with no pattern (and the agreements the node has). */
+  post("/orgs/:orgid/services/:service/search", operation(postServiceSearch)) ({
+    val orgid = params("orgid")
+    // service = svcUrl_svcVersion_svcArch
+    val service = params("service")
+    authenticate().authorizeTo(TNode(OrgAndId(orgid,"*").toString),Access.READ)
+    val resp = response
+    val orgService = "%|"+orgid+"/"+service+"|%"
+    logger.info("SADIYAH: orgService: " + orgService)
+    val q = for {
+      (n, s) <- (NodesTQ.rows.filter(_.orgid === orgid)) join (NodeStatusTQ.rows.filter(_.runningServices like orgService)) on (_.id === _.nodeId)
```

Comment:
For future reference, be aware that if there were more than 1 status resource for a particular node id, then node id would be repeated in your `yield` (and then you would have to remove duplicates before sending them to the client). In this case you are safe, because there can only ever be at most 1 status resource per node.